test/solidity/Facets/UnitFacet.t.sol

```diff
@@ -0,0 +1,684 @@
+    // h1 node public key
+    bytes internal h1NodePublicKey =
+        hex"048633ea6ab7e40cdacf37d1340057e84bb9810de0687af78d031e9b07b65ad4ab379180ab55075f5c2ebb96dab30d2c2fab49d5635845327b6a3c27d20ba4755b";
+    // field node public key
+    bytes internal fieldNodePublicKey =
+        hex"04ae2ab20787f816ea5d13f36c4c4f7e196e29e867086f3ce818abb73077a237f841b33ada5be71b83f4af29f333dedc5411ca4016bd52ab657db2896ef374ce99";
+
+    // Constants for EIP-712
+    // EIP-712 typehash for UnitPayload: keccak256("UnitPayload(bytes32 transactionId,uint256 minAmount,address depositAddress,uint256 destinationChainId,address sendingAssetId,uint256 deadline)");
+    // this is the same as the typehash in the UnitFacet contract
+    bytes32 internal constant UNIT_PAYLOAD_TYPEHASH =
+        0xe08ec0e9d28855df976cf9018cf2d505eaa58b6ebdbf14490f48f2d4c4c13cd3;
+
+    struct UnitPayload {
+        bytes32 transactionId;
+        uint256 minAmount;
+        address depositAddress;
+        uint256 destinationChainId;
+        address sendingAssetId;
+        uint256 deadline;
+    }
+
+    function setUp() public {
+        customBlockNumberForForking = 17130542;
+        initTestBase();
+
+        unitFacet = new TestUnitFacet(backendSignerAddress);
+        bytes4[] memory functionSelectors = new bytes4[](4);
+        functionSelectors[0] = unitFacet.startBridgeTokensViaUnit.selector;
+        functionSelectors[1] = unitFacet
+            .swapAndStartBridgeTokensViaUnit
+            .selector;
+        functionSelectors[2] = unitFacet.addDex.selector;
+        functionSelectors[3] = unitFacet
+            .setFunctionApprovalBySignature
+            .selector;
+
+        addFacet(diamond, address(unitFacet), functionSelectors);
+        unitFacet = TestUnitFacet(address(diamond));
+        // whitelist uniswap dex with function selectors
+        unitFacet.addDex(address(uniswap));
+        unitFacet.addDex(address(unitFacet));
+        unitFacet.setFunctionApprovalBySignature(
+            uniswap.swapExactTokensForTokens.selector
+        );
+        unitFacet.setFunctionApprovalBySignature(
+            uniswap.swapTokensForExactETH.selector
+        );
+        unitFacet.setFunctionApprovalBySignature(
+            uniswap.swapExactTokensForETH.selector
+        );
+        unitFacet.setFunctionApprovalBySignature(
+            uniswap.swapETHForExactTokens.selector
+        );
+
+        setFacetAddressInTestBase(address(unitFacet), "UnitFacet");
+
+        // adjust bridgeData
+        bridgeData.bridge = "unit";
+        bridgeData.destinationChainId = LIFI_CHAIN_ID_HYPERCORE;
+        bridgeData.sendingAssetId = LibAsset.NULL_ADDRESS;
+        bridgeData.minAmount = 0.05 ether; // minimum amount is 0.05 ETH (5e16 wei) mentioned in https://docs.hyperunit.xyz/developers/api/generate-address
+
+        bridgeData.receiver = randomDepositAddress;
```

Comment:
why not USER_RECEIVER?
this is confusing, since the `depositAddress` is usually where we would send the funds to on src chain, not the address that is going to receive the funds